operator/v1/types_csi_cluster_driver.go

```diff
@@ -0,0 +1,78 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterCSIDriver object allows management and configuration of a CSI driver operator
+// installed by default in OpenShift.
+//
+// An example configuration of a CSI driver operator would look like:
+//   apiVersion: operator.openshift.io/v1alpha1
+//   kind: "ClusterCSIDriver"
+//   metadata:
+//     name: "ebs.csi.aws.com-ebs"
+//   spec:
+//     logLevel: Debug
+//     driverConfig:
+//       driverName: "ebs.csi.aws.com-ebs"
+type ClusterCSIDriver struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ClusterCSIDriverSpec `json:"spec"`
+
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status ClusterCSIDriverStatus `json:"status"`
+}
+
+// CSIDriverName is the name of the CSI driver
+type CSIDriverName string
```

Comment:
I think usage like this works: https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L281-L282

Reply:
Added. But still I can't remove the patch file because what we are doing is, we are preventing users from creating arbitrarily named CRs and only following values defined in patch file are permitted.